### PR TITLE
Add AdapterRegistered event log message

### LINKED ISSUE #7038: [CT-2171] [Feature] Add adapter(s) version(s) to the log
It would make things much easier when debugging a dbt invocation, if the adapter version were present.

Right now we get (below for a `dbt debug`)

```
============================== 2023-02-22 23:21:37.421632 | d8954cb4-b0c3-461a-a0f1-68c918c2fe23 ==============================
[0m23:21:37.421632 [info ] [MainThread]: Running with dbt=1.4.1
[0m23:21:37.422456 [debug] [MainThread]: running dbt with arguments {'write_json': True, 'use_colors': True, 'printer_width': 80, 'version_check': True, 'partial_parse': True, 'static_parser': True, 'profiles_dir': '/Users/fleide/.dbt', 'send_anonymous_usage_stats': True, 'quiet': False, 'no_print': False, 'cache_selected_only': False, 'config_dir': False, 'which': 'debug', 'indirect_selection': 'eager'}
...
[0m23:21:38.799040 [debug] [MainThread]: Flushing usage events
[0m23:21:39.154756 [debug] [MainThread]: Connection 'debug' was properly closed.
```

But I could be running that with `dbt-bigquery` 1.4.0, 1.4.1 or 1.4.2!

If we could get that info logeed (anywhere practical, it doesn't need to be at the top):

```
============================== 2023-02-22 23:21:37.421632 | d8954cb4-b0c3-461a-a0f1-68c918c2fe23 ==============================
[0m23:21:37.421632 [info ] [MainThread]: Running with dbt=1.4.1
[0m23:21:37.422456 [debug] [MainThread]: running dbt with arguments {'write_json': True, 'use_colors': True, 'printer_width': 80, 'version_check': True, 'partial_parse': True, 'static_parser': True, 'profiles_dir': '/Users/fleide/.dbt', 'send_anonymous_usage_stats': True, 'quiet': False, 'no_print': False, 'cache_selected_only': False, 'config_dir': False, 'which': 'debug', 'indirect_selection': 'eager'}
...
[0m23:21:38.122456 [info ] [MainThread]: Running with dbt-bigquery=1.4.0
...
[0m23:21:38.799040 [debug] [MainThread]: Flushing usage events
[0m23:21:39.154756 [debug] [MainThread]: Connection 'debug' was properly closed.
```

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
resolves #7038 

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by emmyoop — APPROVED** [review]
LGTM

File: core/dbt/adapters/factory.py [hunk 1]
```diff
@@ -89,7 +89,8 @@ def load_plugin(self, name: str) -> Type[Credentials]:
     def register_adapter(self, config: AdapterRequiredConfig) -> None:
         adapter_name = config.credentials.type
         adapter_type = self.get_adapter_class_by_name(adapter_name)
-
+        adapter_version = import_module(f".{adapter_name}.__version__", "dbt.adapters").version
+        fire_event(AdapterRegistered(adapter_name=adapter_name, adapter_version=adapter_version))
```
Comment (MichelleArk) [inline core/dbt/adapters/factory.py:93]: Took it for a spin, and it looks like there are some formatting inconsistencies between the adapter and dbt versions:

```sh
04:35:56  Running with dbt=1.6.0-b4
04:35:56  target not specified in profile 'postgres', using 'default'
04:35:57  Registered adapter: postgres, version: 1.6.0b4 # no - between 1.6.0 and b4
```

We use [dbt.semver.VersionSpecifier.from_version_string](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/version.py#L38) to [format the core version](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/version.py#L236), for example:  
```python
import dbt.semver
adapter_version_specifier = dbt.semver.VersionSpecifier.from_version_string(adapter_version)
fire_event(AdapterRegistered(adapter_name=adapter_name, adapter_version=str(adapter_version_specifier)))
```

File: core/dbt/events/types.py [hunk 2]
```diff
@@ -655,6 +655,14 @@ def message(self) -> str:
 # Skipping E032, E033, E034
 
 
+class AdapterRegistered(InfoLevel):
+    def code(self):
+        return "E034"
+
+    def message(self) -> str:
+        return f"Registered adapter: {self.adapter_name}={self.adapter_version}"
```
Comment (MichelleArk) [inline core/dbt/events/types.py:663]: One last formatting issue with an extra `=`. not entirely sure where it's coming from but perhaps we need to get rid the extra one in this message?

```
17:10:01  Running with dbt=1.6.0-b4
17:10:01  target not specified in profile 'postgres', using 'default'
17:10:01  Registered adapter: postgres==1.6.0-b4
```

File: core/dbt/adapters/factory.py [hunk 3]
```diff
@@ -89,7 +90,11 @@ def load_plugin(self, name: str) -> Type[Credentials]:
     def register_adapter(self, config: AdapterRequiredConfig) -> None:
         adapter_name = config.credentials.type
         adapter_type = self.get_adapter_class_by_name(adapter_name)
-
+        adapter_version = import_module(f".{adapter_name}.__version__", "dbt.adapters").version
+        adapter_version_specifier = VersionSpecifier.from_version_string(adapter_version).to_version_string()
```
Comment (MichelleArk) [inline core/dbt/adapters/factory.py:94]: non-blocking suggestion: Assuming there are no circular dependencies, what do you think about passing the 'raw' `adapter_version` string directly to the `AdapterRegistered` event and using `VersionSpecifier` within the event to format the string in the message? That way formatting concerns are handled closer to the message relying on them.

File: core/dbt/events/types.py (hunk 2 above)
Comment (colin-rogers-dbt) [reply core/dbt/events/types.py]: hah didn't realize that our version specifier adds an `=` prefix

**Review by MichelleArk — APPROVED** [review]

File: core/dbt/adapters/factory.py (hunk 3 above)
Comment (colin-rogers-dbt) [reply core/dbt/adapters/factory.py:94]: I agree it would be cleaner but looks like it does introduce a circular dependency

Comment (MichelleArk) [reply core/dbt/adapters/factory.py:94]: dang!

**Comment by jtcohen6** [comment]
@Fleid Would you want to backport this for inclusion in next v1.5.x patch? It's a new thing, but:
- the code change footprint is quite small
- we've wanted this for over a year (#5124 + #7038)
- it would be immediately helpful for users (& our Support team) debugging

**Pull request merged**